coalib/tests/processes/communication/InterruptProcessTest.py

```diff
@@ -0,0 +1,56 @@
+import unittest
+import sys
+import time
+import multiprocessing
+
+sys.path.insert(0, ".")
+from coalib.processes.communication.InterruptProcess import interrupt_process
+
+
+def runner(queue):
+    try:
+        while True:
+            queue.put("Hello coala", timeout=0.1)
+            time.sleep(0.8)
+    finally:
+        queue.put("End", timeout=0.1)
+
+
+def runner2(queue):
+    try:
+        runner(queue)
+    finally:
+        queue.put("End2", timeout=0.1)
+
+
+class InterruptProcessTest(unittest.TestCase):
+    def test_interrupt_process(self):
+        queue = multiprocessing.Queue()
+        p = multiprocessing.Process(target=runner, kwargs={'queue': queue})
+        p.start()
+        time.sleep(2)
+        interrupt_process(p.pid)
+        self.assertEqual("Hello coala", queue.get(timeout=0.1))
+        self.assertEqual("Hello coala", queue.get(timeout=0.1))
+        self.assertEqual("Hello coala", queue.get(timeout=0.1))
+        self.assertEqual("End", queue.get(timeout=0.1))
+        p.terminate()
+        p.join()
+
+    def test_interrupt_process_with_nested_functions(self):
+        queue = multiprocessing.Queue()
+        p = multiprocessing.Process(target=runner2, kwargs={'queue': queue})
+        p.start()
+        time.sleep(2)
+        interrupt_process(p.pid)
+        self.assertEqual("Hello coala", queue.get(timeout=0.1))
+        self.assertEqual("Hello coala", queue.get(timeout=0.1))
+        self.assertEqual("Hello coala", queue.get(timeout=0.1))
+        self.assertEqual("End", queue.get(timeout=0.1))
+        self.assertEqual("End2", queue.get(timeout=0.1))
+        p.terminate()
+        p.join()
+
+
+if __name__ == '__main__':
+    unittest.main(verbosity=2)
```

Comment:
I wonder if on windows, you could split off a new process group here in which you execute the test? This would be compatible with our testing infrastructure only for now but pytest is no priority so I'd be fine with that.